protocol/src/common/protocol.ts

```diff
@@ -3448,10 +3448,34 @@ export interface DocumentRangeFormattingParams extends WorkDoneProgressParams {
 	options: FormattingOptions;
 }
 
+/**
+ * The parameters of a {@link DocumentRangesFormattingRequest}.
+ */
+export interface DocumentRangesFormattingParams extends WorkDoneProgressParams {
+	/**
+	 * The document to format.
+	 */
+	textDocument: TextDocumentIdentifier;
+
+	/**
+	 * The ranges to format
+	 */
+	ranges: Range[];
+
+	/**
+	 * The format options
+	 */
+	options: FormattingOptions;
+}
+
 /**
  * Provider options for a {@link DocumentRangeFormattingRequest}.
  */
 export interface DocumentRangeFormattingOptions extends WorkDoneProgressOptions {
+	/**
+     * Whether the server supports formatting multiple ranges at once.
+     */
+	canFormatMultipleRanges?: boolean;
```

Comment:
to make things consitent with other name and the client capability I would name it `multipleRangesSupport` or `rangesSupport`

Reply:
Done.